networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManager.java

```diff
@@ -91,17 +98,37 @@ protected SafeFuture<?> doStart() {
     synchronized (this) {
       staticPeers.forEach(this::createPersistentConnection);
     }
-    periodicPeerSearch =
-        asyncRunner.runWithFixedDelay(
-            this::searchForPeers,
-            DISCOVERY_INTERVAL,
-            error -> LOG.error("Error while searching for peers", error));
-    connectToBestPeers(emptyList());
-    searchForPeers();
+    asyncRunner.runAsync(this::createRecurrentSearchTask).ifExceptionGetsHereRaiseABug();
```

Comment:
Why are we running this async? The `searchForPeers` done in `createRecurrentSearchTask` already starts an async process and returns quickly.

Reply:
Nice point